src/frontend/app/features/service-catalog/services.service.ts

```diff
@@ -49,14 +56,107 @@ export class ServicesService {
+        paginationMonitor: this.paginationMonitorFactory.create(
+          paginationKey,
+          entityFactory(servicePlanVisibilitySchemaKey)
+        )
+      },
+      true
+    ).entities$;
+  }
+  getServicePlanVisibilitiesForPlan = (servicePlanGuid: string): Observable<APIResource<IServicePlanVisibility>[]> => {
+    return this.servicePlanVisibilities$.pipe(
+      filter(p => !!p),
+      map(vis => vis.filter(s => s.entity.service_plan_guid === servicePlanGuid)),
+      first()
+    );
+  }
 
+  getVisiblePlans = () => {
+    return this.servicePlans$.pipe(
+      filter(p => !!p && p.length > 0),
+      map(o => o.filter(s => s.entity.bindable)),
+      combineLatest(this.servicePlanVisibilities$),
+      map(([svcPlans, svcPlanVis]) => this.fetchVisiblePlans(svcPlans, svcPlanVis)),
+    );
   }
 
+  fetchVisiblePlans =
+    (svcPlans: APIResource<IServicePlan>[], svcPlanVis: APIResource<IServicePlanVisibility>[]): APIResource<IServicePlan>[] => {
+      const visiblePlans: APIResource<IServicePlan>[] = [];
+      svcPlans.forEach(p => {
+        if (p.entity.public) {
+          visiblePlans.push(p);
+        } else if (svcPlanVis.filter(svcVis => svcVis.entity.service_plan_guid === p.metadata.guid).length > 0) {
+          visiblePlans.push(p);
+        }
+      });
+      return visiblePlans;
+    }
+
+  getServicePlanByGuid = (servicePlanGuid: string) => {
+    return this.servicePlans$.pipe(
+      filter(p => !!p),
+      map(servicePlans => servicePlans.filter(o => o.metadata.guid === servicePlanGuid)),
+      map(o => o[0])
+    );
+  }
+
+  getSelectedServicePlan = (): Observable<APIResource<IServicePlan>> => {
+    return Observable.combineLatest(this.store.select(selectCreateServiceInstanceServicePlan), this.servicePlans$)
+      .pipe(
+        filter(([p, q]) => !!p && !!q),
+        map(([servicePlanGuid, servicePlans]) => servicePlans.filter(o => o.metadata.guid === servicePlanGuid)),
+        map(p => p[0]),
+        filter(p => !!p)
+      );
+  }
+
+  getOrgsForSelectedServicePlan = (): Observable<APIResource<IOrganization>[]> => {
+    return this.getSelectedServicePlan()
+      .pipe(
+        switchMap(servicePlan => {
+          if (servicePlan.entity.public) {
+            const getAllOrgsAction = CloudFoundryEndpointService.createGetAllOrganizationsLimitedSchema(this.cfGuid);
+            return getPaginationObservables<APIResource<IOrganization>>({
+              store: this.store,
+              action: getAllOrgsAction,
+              paginationMonitor: this.paginationMonitorFactory.create(
+                getAllOrgsAction.paginationKey,
+                entityFactory(organizationSchemaKey)
+              )
+            }, true)
+              .entities$.pipe(
+                share(),
+                first()
+              );
+          } else {
+            // Service plan is not public, fetch visibilities
+            return this.getServicePlanVisibilitiesForPlan(servicePlan.metadata.guid)
```

Comment:
I think there's a bug here. If you refresh on any service page and then try to create a non-public plan the org entities spaces param is not populated and leads to the following error
```
ERROR TypeError: Cannot read property 'length' of undefined
    at eval (specify-details-step.component.ts:137)
```
It looks like the root cause it the GetServicePlanVisibilities action which fetches the org is missing the `spaces` relationship (in the relations and possibly schema).